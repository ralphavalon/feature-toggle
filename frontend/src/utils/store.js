import React, { useState } from 'react';

export const StoreContext = React.createContext(null);

export default ({ children }) => {
  const [features, setFeatures] = useState([]);
  const [hasError, setHasError] = useState(false);

  const store = {
    features: [features, setFeatures],
    error: [hasError, setHasError]
  };
  return (
    <StoreContext.Provider value={store}>
      {children}
    </StoreContext.Provider>
  );
};

