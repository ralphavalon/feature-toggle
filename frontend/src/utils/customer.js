import React, { useContext } from 'react';
import { StoreContext } from './store';

const getCustomerIds = () => {
  const { features: [features, setFeatures] } = useContext(StoreContext);

  let customerIds = [];
  features.forEach(f => {
    customerIds = customerIds.concat(f.customerIds.filter((item) => customerIds.indexOf(item) < 0));
  });
  return customerIds;
};

export { getCustomerIds };
