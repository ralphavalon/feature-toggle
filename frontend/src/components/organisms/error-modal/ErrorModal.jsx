import React, { useContext } from 'react';
import { Button, Modal } from 'react-bootstrap';
import { StoreContext } from '../../../utils/store';

const ErrorModal = () => {

  const { error: [hasError, setHasError] } = useContext(StoreContext);

  const handleClose = () => setHasError(false);

  return (
    <Modal show={hasError} onHide={handleClose}>
      <Modal.Header>
        <Modal.Title>Something went wrong!</Modal.Title>
      </Modal.Header>
      <Modal.Body>Unable to complete the request. Please check your data and try again later.</Modal.Body>
      <Modal.Footer>
        <Button variant="outline-danger" onClick={handleClose}>Close</Button>
      </Modal.Footer>
    </Modal>
  );
};

export default ErrorModal;
