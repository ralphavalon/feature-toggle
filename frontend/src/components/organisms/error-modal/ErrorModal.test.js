import React from 'react';
import { shallow } from 'enzyme';

import { StoreContext } from '../../../utils/store';

import ErrorModal from './ErrorModal';

describe('ErrorModal', () => {
  it('renders correctly', () => {
    let wrapper = shallow(
      <StoreContext.Provider value={{
        error: [false, jest.fn()]
      }}>
        <ErrorModal />
      </StoreContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();

    wrapper = shallow(
      <StoreContext.Provider value={{
        error: [true, jest.fn()]
      }}>
        <ErrorModal />
      </StoreContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
})

