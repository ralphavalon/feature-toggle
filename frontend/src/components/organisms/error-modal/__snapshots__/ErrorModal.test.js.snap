// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal renders correctly 1`] = `<ErrorModal />`;

exports[`ErrorModal renders correctly 2`] = `<ErrorModal />`;
